src/time-resolution.jl

```diff
@@ -13,86 +13,91 @@ which specifies the partition strategy for each resolution (i.e., lowest or high
         (
             name = :lowest_storage_level_intra_rp,
             partitions = _all,
             strategy = :lowest,
-            asset_filter = a -> graph[a].type == "storage" && !graph[a].is_seasonal,
+            asset_filter = (a, y) ->
+                graph[a].type == "storage" && !get(graph[a].is_seasonal, y, false),
         ),
         (
             name = :lowest_in_out,
             partitions = _allflows,
             strategy = :lowest,
-            asset_filter = a ->
-                graph[a].type == "storage" && !ismissing(graph[a].use_binary_storage_method),
+            asset_filter = (a, y) ->
+                graph[a].type == "storage" &&
+                    !ismissing(get(graph[a].use_binary_storage_method, y, missing)),
         ),
         (
             name = :highest_in_out,
             partitions = _allflows,
             strategy = :highest,
-            asset_filter = a -> graph[a].type in ["hub", "consumer"],
+            asset_filter = (a, y) -> graph[a].type in ["hub", "consumer"],
         ),
         (
             name = :highest_in,
             partitions = _inflows,
             strategy = :highest,
-            asset_filter = a -> graph[a].type in ["storage"],
+            asset_filter = (a, y) -> graph[a].type in ["storage"],
         ),
         (
             name = :highest_out,
             partitions = _outflows,
             strategy = :highest,
-            asset_filter = a -> graph[a].type in ["producer", "storage", "conversion"],
+            asset_filter = (a, y) -> graph[a].type in ["producer", "storage", "conversion"],
         ),
         (
             name = :units_on,
             partitions = _assets,
             strategy = :highest,
-            asset_filter = a ->
-                graph[a].type in ["producer", "conversion"] && graph[a].unit_commitment,
+            asset_filter = (a, y) ->
+                graph[a].type in ["producer", "conversion"] && graph[a].unit_commitment[y],
         ),
         (
             name = :units_on_and_outflows,
             partitions = _assets_and_outflows,
             strategy = :highest,
-            asset_filter = a ->
-                graph[a].type in ["producer", "conversion"] && graph[a].unit_commitment,
+            asset_filter = (a, y) ->
+                graph[a].type in ["producer", "conversion"] && graph[a].unit_commitment[y],
         ),
     ]
 
-    num_rep_periods = length(representative_periods)
+    RP = Dict(year => 1:length(representative_periods[year]) for year in years)
 
     for (name, partitions, strategy, asset_filter) in partitions_cases
         constraints_partitions[name] = OrderedDict(
-            (a, rp) => begin
-                P = partitions(a, rp)
+            (a, y, rp) => begin
+                P = partitions(a, y, rp)
                 if length(P) > 0
-                    compute_rp_partition(partitions(a, rp), strategy)
+                    compute_rp_partition(partitions(a, y, rp), strategy)
```

Comment:
Here's another question:

I had before something like this:

```
result = compute_rp_partition(partitions(a, y, rp), strategy)
result
```

This is because when I didn't add this explicit output, it gave me an error...do you have any thoughts here why that could happen?

Reply:
Not really, it shouldn't be the case. Both codes should do the same thing, as far as I can see, which is why I removed the change.